### PR TITLE
rustc: improve E0669 span

### PR DESCRIPTION
**Pull request by levex**
E0669 refers to an operand that cannot be coerced into a single LLVM
value, unfortunately right now this uses the Span for the entire inline
assembly statement, which is less than ideal.

This commit preserves the Span from HIR, which lets us emit the error
using the Span for the operand itself in MIR.

r? @nagisa 
cc/ @parched

### CODE REVIEW

File: src/librustc/mir/visit.rs [hunk 1]
```diff
@@ -383,7 +383,7 @@ macro_rules! make_mir_visitor {
                         for output in & $($mutability)* outputs[..] {
                             self.visit_place(output, PlaceContext::AsmOutput, location);
                         }
-                        for input in & $($mutability)* inputs[..] {
+                        for (_, input) in & $($mutability)* inputs[..] {
                             self.visit_operand(input, location);
```
Comment (nagisa) [inline src/librustc/mir/visit.rs]: This needs a `self.visit_span`.

Comment (levex) [reply src/librustc/mir/visit.rs]: Fixed. I've force pushed to avoid another CI check when I squash the commits...

**Comment by levex** [comment]
Just a heads up: I will finish this PR this week, but I've been busy recently so I haven't had the time yet. It's not forgotten about. :-)

File: src/librustc_codegen_llvm/mir/statement.rs [hunk 2]
```diff
@@ -84,18 +84,18 @@ impl FunctionCx<'a, 'll, 'tcx> {
                 }).collect();
 
                 let input_vals = inputs.iter()
-                    .try_fold(Vec::with_capacity(inputs.len()), |mut acc, input| {
+                    .try_fold(Vec::with_capacity(inputs.len()), |mut acc, (span, input)| {
```
[the remark this answers is not in the document]
Comment (estebank) [reply src/librustc_codegen_llvm/mir/statement.rs]: @parched we prefer to show as many as possible, as long as they are _relevant_.

[the remark this answers is not in the document]
Comment (levex) [reply src/librustc_codegen_llvm/mir/statement.rs]: Fixed in 32ee20f

**Review by parched — COMMENTED** [review]
Looks good. It's probably worth adding a test for the multiple error case now though.

**Comment by levex** [comment]
@parched done!

r? @nagisa

**Pull request merged**